sklearn/utils/validation.py

```diff
@@ -1655,6 +1655,51 @@ def check_is_fitted(estimator, attributes=None, *, msg=None, all_or_any=all):
         raise NotFittedError(msg % {"name": type(estimator).__name__})
 
 
+def _estimator_has(attr, delegates=None):
```

Comment:
I think that we can impose to have keyword only argument for the new parameters. It will enforce readibility.

```suggestion
def _estimator_has(attr, *, delegates=None):
```

Reply:
I had to think about the meaning of `None` here. I think that we can use a tuple instead for the default and it will be more understantable:


```suggestion
def _estimator_has(attr, *, delegates=("estimator_", "estimator")):
```

Since this is an immutable default, we will not have any issue.